keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,165 @@
+        training: A boolean argument that determines whether the call should be run 
+            in inference mode or training mode. Default: True.
+   
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    rgb_shift = keras_cv.layers.RGBShift(factor=(-2, 2))
+    augmented_images = rgb_shift(images)
+    ```
+    """
+
+    _FACTOR_VALIDATION_ERROR = (
+        "The factor should be a scalar, "
+        "a tuple or a list of two upper and lower "
+        "bound values in the range `(-1.0, 1.0)` as float or `(-255, 255) as integer."
+    )
+
+    def __init__(self, factor, seed=None, **kwargs):
+        super().__init__(**kwargs)
+        self.factor = self._set_factor_limit(factor)
+        self.seed = seed
+
+    def _set_factor_limit(self, factor):
+        if isinstance(factor, (tuple, list)):
+            if len(factor) != 2:
+                raise ValueError(
+                    self._FACTOR_VALIDATION_ERROR + f" Received: factor={factor}"
+                )
+            return self._check_factor_range(sorted(factor))
+        elif isinstance(factor, (int, float)):
+            factor = abs(factor)
+            return self._check_factor_range([-factor, factor])
+        else:
+            raise ValueError(
+                self._FACTOR_VALIDATION_ERROR + f" Received: factor={factor}"
+            )
+
+    def _check_factor_range(self, factor):
+        if all(isinstance(each_elem, float) for each_elem in factor):
+            if factor[0] < -1.0 or factor[1] > 1.0:
+                raise ValueError(
+                    self._FACTOR_VALIDATION_ERROR + f" Received: factor={factor}"
+                )
+            return factor
+        elif all(isinstance(each_elem, int) for each_elem in factor):
+            if factor[0] < -255 or factor[1] > 255:
+                raise ValueError(
+                    self._FACTOR_VALIDATION_ERROR + f" Received: factor={factor}"
+                )
+            return factor
+        else:
+            raise ValueError(
+                "Both lower/upper bound values must have the same dtype. "
+                f"Received: factor={factor} where type(factor[0]) is {type(factor[0])} "
+                f"and type(factor[1]) is {type(factor[1])}"
+            )
+
+    def _get_random_uniform(self, factor_limit, rgb_delta_shape):
+        if self.seed is not None:
+            _rand_uniform = tf.random.stateless_uniform(
+                shape=rgb_delta_shape,
+                seed=[0, self.seed],
+                minval=factor_limit[0],
+                maxval=factor_limit[1],
+            )
+        else:
+            _rand_uniform = tf.random.uniform(
+                rgb_delta_shape,
+                minval=factor_limit[0],
+                maxval=factor_limit[1],
+                dtype=tf.float32,
+            )
+
+        if all(isinstance(each_elem, float) for each_elem in factor_limit):
+            _rand_uniform = _rand_uniform * 85.0
```

Comment:
I hadn't caught that initially, but why 85? Why not 255?

Reply:
When the input factor is a **float** of ranges `[-1.0, 1.0]`, each `_rand_uniform` of r, g, and b channel get scaled up by **1/3th** of `255.0` (or, `85.0`). 

I think it's needed as the expected input ranges are `[0-255]`. For **integer** factor of `[0, 255]` ranges, the `_rand_uniform` is not required to scale up.

Reply:
lets always assume [0, 255] as the value range, with an optional `value_range` parameter that performs a transformation on the images at the start of the `augment_image` call.